plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/ExpressionConverter.java

```diff
@@ -115,102 +110,106 @@ private static Expression toIcebergExpression(String columnName, Type type, Doma
-                                lessThanOrEqual(columnName, getIcebergLiteralValue(type, high)));
-                        expression = or(expression, between);
-                    }
-                }
-                else {
-                    if (lowBound == EXACTLY && low.getValueBlock().isPresent()) {
-                        // case >=
-                        expression = or(expression, greaterThanOrEqual(columnName, getIcebergLiteralValue(type, low)));
-                    }
-                    else if (lowBound == ABOVE && low.getValueBlock().isPresent()) {
-                        // case >
-                        expression = or(expression, greaterThan(columnName, getIcebergLiteralValue(type, low)));
-                    }
-
-                    if (highBound == EXACTLY && high.getValueBlock().isPresent()) {
-                        // case <=
-                        if (low.getValueBlock().isPresent()) {
-                            expression = and(expression, lessThanOrEqual(columnName, getIcebergLiteralValue(type, high)));
-                        }
-                        else {
-                            expression = or(expression, lessThanOrEqual(columnName, getIcebergLiteralValue(type, high)));
-                        }
-                    }
-                    else if (highBound == BELOW && high.getValueBlock().isPresent()) {
-                        // case <
-                        if (low.getValueBlock().isPresent()) {
-                            expression = and(expression, lessThan(columnName, getIcebergLiteralValue(type, high)));
-                        }
-                        else {
-                            expression = or(expression, lessThan(columnName, getIcebergLiteralValue(type, high)));
-                        }
-                    }
-                }
+                expression = or(expression, toIcebergExpression(columnName, range));
             }
             return expression;
         }
 
         throw new VerifyException("Did not expect a domain value set other than SortedRangeSet but got " + domainValues.getClass().getSimpleName());
     }
 
-    private static Object getIcebergLiteralValue(Type type, Marker marker)
+    private static Expression toIcebergExpression(String columnName, Range range)
     {
+        Type type = range.getType();
+
+        if (range.isSingleValue()) {
+            Object icebergValue = getIcebergLiteralValue(type, range.getSingleValue());
+            return equal(columnName, icebergValue);
+        }
+
+        if (range.isLowInclusive() && range.isHighInclusive()) {
+            Object icebergLow = getIcebergLiteralValue(type, range.getLowBoundedValue());
+            Object icebergHigh = getIcebergLiteralValue(type, range.getHighBoundedValue());
+
+            return and(
+                    greaterThanOrEqual(columnName, icebergLow),
+                    lessThanOrEqual(columnName, icebergHigh));
+        }
```

Comment:
why do we need this? It seems handled below.

Reply:
You're right. This was special-cased in the previous code, but is not needed, will remove.